Add recipe modin

devin-petersohn opened the pull request:
<!--
Thank you very much for putting in this recipe PR!

This repository is very active, so if you need help with
a PR or once it's ready for review, please let the right people know.
There are language-specific teams for reviewing recipes.

Currently available teams are:
- python `@conda-forge/help-python`
- python/c hybrid `@conda-forge/help-python-c`
- r `@conda-forge/help-r`
- java `@conda-forge/help-java`
- nodejs `@conda-forge/help-nodejs`
- c/c++ `@conda-forge/help-c-cpp`
- perl `@conda-forge/help-perl`
- Julia `@conda-forge/help-julia`
- ruby `@conda-forge/help-ruby`

If your PR doesn't fall into those categories please contact
the full review team `@conda-forge/staged-recipes`.

Due to GitHub limitations first time contributors to conda-forge are unable
to ping these teams. You can [ping the team](https://conda-forge.org/docs/maintainer/infrastructure.html#conda-forge-admin-please-ping-team) using a special command in
a comment on the PR to get the attention of the `staged-recipes` team. You can
also consider asking on our [Gitter channel](https://gitter.im/conda-forge/conda-forge.github.io)
or on our [Keybase chat](https://keybase.io/team/condaforge.chat)
if your recipe isn't reviewed promptly.
-->

cc @h-vetinari 

Derived and rebased from #8471. 

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/modin```) and found it was in an excellent condition.

devin-petersohn commented:
@h-vetinari I updated the source and version, reset to PyPI because that seems to be working now. I also squashed to a single commit after all tests pass. Please verify that you are still willing to be a recipe maintainer for review purposes.

synapticarbors commented on recipes/modin/meta.yaml:
```diff
@@ -0,0 +1,43 @@
+build:
+  number: 0
+  script: "{{ PYTHON }} -m pip install . --no-deps --ignore-installed -vv "
+  skip: true  # [py<35]
```
I would suggest removing the `skip` line and make this package `noarch: python`, and then change `python` to `python >=3.5` under `host` and `run`

synapticarbors commented on recipes/modin/LICENSE:
```diff
@@ -0,0 +1,201 @@
+                                 Apache License
```
A license header is included in every source file in the package, so I'm not sure about whether we need to manually package the license file. Then the question would be how to specify the `license_file` below. I'm not sure best how to handle this.

synapticarbors commented on recipes/modin/meta.yaml:
```diff
@@ -0,0 +1,43 @@
+
+test:
+  imports:
+    - modin
```
Also maybe do some imports of submodules. e.g.:

```
- modin.pandas
- modin.sql
- modin.data_management.functions
...
```

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/modin```) and found some lint.

Here's what I've got...


For **recipes/modin**:

 * Non noarch packages should have python requirement without any version constraints.
 * Non noarch packages should have python requirement without any version constraints.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/modin```) and found it was in an excellent condition.

devin-petersohn commented:
New Tests pass, version string formatted correctly, simple dataframe created correctly.

@synapticarbors Please let me know if there is some additional change needed for the license file.
